packages/tests/src/utils/withGracefulTimeout.ts

```diff
@@ -0,0 +1,38 @@
+import { Logger } from "@waku/utils";
+const log = new Logger("test:mocha-hook");
+
+export function withGracefulTimeout(
```

Comment:
to decrease amount of lines changed - I want to propose following:
```ts
export const beforeEachCustom = function (self, cb: Promise<void>, setter: any) {
  self.beforeEach((done) => {
    this.timeout(MOCHA_HOOK_MAX_TIMEOUT);
    withGracefulTimeout(async () => {
      const [nwaku, waku] = await runMultipleNodes(
        this,
        [DefaultPubsubTopic],
        strictCheckNodes
      );
      subscription = await waku.filter.createSubscription();
      
      setter({ nwaku, waku/*, nwaku1, ...etc */});
    }, 20000, done);
  });
});
```

do you think it would work, @fbarbu15 ?

Reply:
I did made it work but there are multiple functions that we need to wrap:
```
runMultipleNodes 
runNodes (2 of them)
tearDownNodes
teardownNodesWithRedundancy
```
and direct usage of
`nwaku.start + other commands like subscribe `

So we would probably need to create multiple customFunctions (we will complicate stuff even further) or unify the mentioned methods to have only one general runNodes function and only one general tearDownNodes function (this one looks like a big refactoring but it may be worth it) 
Wdyt?

Reply:
I think this is solvable the same way: 

in utils file:
```ts
export const beforeEachCustom = function (self, cb: Promise<void>) {
  self.beforeEach((done) => {
    this.timeout(MOCHA_HOOK_MAX_TIMEOUT);
    withGracefulTimeout(cb, 20000, done);
  });
});

export const afterEachCustom = function (self, cb: Promise<void>, setter: any) {
  self.afterEach((done) => {
    this.timeout(MOCHA_HOOK_MAX_TIMEOUT);
    withGracefulTimeout(cb, 20000, done);
  });
});
```

in test file:
```ts

describe("test", () => {
  let nwaku, waku;

  beforeEachCustom(async () => {
    [nwaku, waku] = await runMultipleNodes(
      this,
      [DefaultPubsubTopic],
      strictCheckNodes
    );
    subscription = await waku.filter.createSubscription();
  });

  afterEachCustom(async () => {
    await teardownNodesWithRedundancy(serviceNodes, waku);
  });

  test("test case", () => {});
});
```